subsys/bluetooth/controller/hci/hci_driver.c

```diff
@@ -734,13 +733,17 @@ static int hci_driver_open(void)
 
 	DEBUG_INIT();
 
+	k_fifo_init(&recv_fifo);
+	k_sem_init(&sem_prio_recv, 0, K_SEM_MAX_LIMIT);
+
 	err = ll_init(&sem_prio_recv);
```

Comment:
`lll_init()` requests LF clock on, multiple `open()` could accumulate the reference count in the clock_control driver?

Either `lll_init()` has to be moved outside of `open()` to be only at power on reset or introduce a `lll_deinit()` which is a combination of `lll_reset()` plus a new `lll_clock_deinit()`  so that LF clock request is released.

Reply:
If something needs to be run just once, could they be put in `hci_driver_init`?

Reply:
>Either lll_init() has to be moved outside of open() to be only at power on reset or introduce a lll_deinit() which is a combination of lll_reset() plus a new lll_clock_deinit() so that LF clock request is released.

If we did this, then `open()` would do nothing LL-related right?

Reply:
> If we did this, then `open()` would do nothing LL-related right?

Yes, and calls to only `ll_reset()` will suffice inside the `close` interface.